src/MongoDB.Driver.Core/Core/Authentication/External/HttpClientHelper.cs

```diff
@@ -0,0 +1,61 @@
+﻿/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Core.Authentication.External
+{
+    internal class HttpClientHelper
+    {
+        #region static
+        public static HttpClient CreateHttpClient() => new HttpClient() { Timeout = TimeSpan.FromSeconds(10) };
+        #endregion
+
+        private readonly HttpClient _httpClient;
+
+        public HttpClientHelper() : this(CreateHttpClient())
+        { }
+
+        internal HttpClientHelper(HttpClient httpClient)
+        {
+            _httpClient = Ensure.IsNotNull(httpClient, nameof(httpClient));
+        }
+
+        public async Task<string> GetHttpContentAsync(HttpRequestMessage request, string exceptionMessage, CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response;
+            string content = null;
+            try
+            {
+                response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
+                return content;
+            }
+            catch (Exception ex) when (ex is OperationCanceledException or HttpRequestException)
+            {
+                if (content != null)
+                {
+                    exceptionMessage = $"{exceptionMessage}. Response body: {content}.";
```

Comment:
initial logic can be found here: https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Authentication/MongoAWSAuthenticator.cs#L366-L381
Some notes:
1. Now spec for GCP requires providing a server response in the result if something goes wrong. Aws didn't have such requirement. But I would prefer to have the same httpClient related logic, so it's a minor (in my opinion) BC, thoughts? 
2. Initial logic had a bug related to handling exception. I think it's related to this discussion: https://github.com/MikalaiMazurenka/mongo-csharp-driver/pull/23/commits/dde784d0a6a619f085068096795fe927fb39058c#r395811513 which was interpreted incorrectly

Reply:
Makes sense. I'd rather surface any failed HTTP status codes to make debugging easier regardless of cloud provider.